From a42db12854dcf7af7ff17b66eca3b7a210baab65 Mon Sep 17 00:00:00 2001
From: Arulanbu Balusamy <quic_abalusam@quicinc.com>
Date: Mon, 17 Jul 2023 19:40:37 +0530
Subject: [PATCH] disable the rddm notifier for kernel 6.x

disable the rddm notifier for 6.x kernel verson, Because 6.1x kernel has no
support for panic notifier.

Signed-off-by: Arulanbu Balusamy <quic_abalusam@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 97f12fe..dd7d6e7 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -3632,6 +3632,8 @@ void *ath12k_get_ab_nd_peer_from_peer_mac(u8 *peer_mac, struct ath12k_base *ab)
 
 #endif /* CPTCFG_ATH12K_SAWF */
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
+
 static void ath12k_rddm_notifier(struct notifier_block *this,
 				 unsigned long event, void *ptr)
 {
@@ -3659,6 +3661,8 @@ static struct notifier_block rddm_notifier = {
 	.notifier_call  = ath12k_rddm_notifier,
 };
 
+#endif 
+
 static int ath12k_init(void)
 {
 	int ret;
@@ -3668,10 +3672,12 @@ static int ath12k_init(void)
 		pr_err("Failed to register debugfs ath12k driver: %d\n", ret);
 		return ret;
 	}
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 
 	/* Add notifier to an atomic notifier chain to get the callback during BUG_ON
 	 */
 	atomic_notifier_chain_register(&panic_notifier_list, &rddm_notifier);
+#endif
 
 	ret = ath12k_ahb_init();
 	if (ret) {
@@ -3699,7 +3705,9 @@ error:
 
 static void ath12k_exit(void)
 {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 	atomic_notifier_chain_unregister(&panic_notifier_list, &rddm_notifier);
+#endif
 	ath12k_pci_exit();
 	ath12k_ahb_exit();
 	ath12k_debugfs_destroy();
-- 
2.17.1

